### ISSUE TITLE
Throttle/Group multiple notifications sent from the same app

### ISSUE THREAD

**Issue by danrabbit** [elementary/notifications#72]
It's not unusual to receive a few notifications at the same time from a Slack bot, for example. Instead of flooding the screen with notifications, it would probably make sense to group them together and just send one notification like:

**Slack**
10 new notifications

We're currently doing something similar in Mail when multiple emails come in at the same time. Making this a feature of the notification server could mean that you'd only get one notification on screen but we could send all the of the individual notifications to the indicator which would be nice. This would resolve the issue where you can end up with something in the indicator that looks like:

**Mail** now
5 unread messages

**Mail** 3pm
New message from Cassidy

**Mail** 2pm
3 unread messages

**Comment by cassidyjames** [comment]
Maybe throttling/grouping in bubbles when sent within a _very_ short timeframe makes sense, but we need to be careful not to accidentally miss really important context. I.e. I would probably expect every individual text message to come in as a bubble, or to be appended to one another or something.

It looks like Android (at least the Messages app) does a "replace" when messages come in within a short timeframe, so you can still see all the text as it comes in, but don't get overwhelmed with individual messages.

**Comment by cassidyjames** [comment]
Example of Messaging notifications on Android. Note that these are individual notifications coming in, because each can be interacted with separately later from the notification center.

![ezgif com-video-to-gif](https://user-images.githubusercontent.com/611168/78842095-57ad2900-79bc-11ea-89a8-8c2506bf73f7.gif)

**Comment by danrabbit** [comment]
Yeah I was thinking only when they are sent within like 1s or something super tight. Like tighter than is reasonable for a human to be sending
